File: JavaTest/src/com/io/stack/TimeUtils.java
package com.io.stack;

import java.text.SimpleDateFormat;
import java.util.Date;

/**
 * author: King.Z <br>
 * date:  2018/2/13 15:00 <br>
 * description: XXXXXXX <br>
 */
public class TimeUtils {
    private static SimpleDateFormat format = new SimpleDateFormat("yyyyMMddHHmmss");
    private static long TimeDelta = System.currentTimeMillis() - timestamp();

    public static long currentTimeMillis() {
        return timestamp() + TimeDelta;
    }

    private static long timestamp() {
        return System.nanoTime() / 1000000L;
    }

    public static String getCurrentTime() {
        try {
            return format.format(new Date(currentTimeMillis()));
        } catch (Exception var1) {
            var1.printStackTrace();
            return null;
        }
    }
}

File: KotlinTest/src/main/kotlin/com/kingz/kt/utils/TimeUtils.kt
package com.kingz.kt.utils

import java.text.SimpleDateFormat
import java.util.*

/**
 * All rights reserved. <br></br>
 * author: King.Z <br></br>
 * date:  2018/2/13 15:00 <br></br>
 * description: XXXXXXX <br></br>
 */
object TimeUtils {
    private val format = SimpleDateFormat("HH:mm:ss")
    private val TimeDelta = System.currentTimeMillis() - timestamp()

    val currentTime: String?
        get() {
            return try {
                format.format(Date(currentTimeMillis()))
            } catch (var1: Exception) {
                var1.printStackTrace()
                null
            }
        }

    private fun currentTimeMillis(): Long {
        return timestamp() + TimeDelta
    }

    private fun timestamp(): Long {
        return System.nanoTime() / 1_000_000L
    }
}
